tests/solana.rs

```diff
@@ -1661,11 +1681,18 @@ impl VirtualMachine {
             &[]
         };
 
-        let func = &self.stack[0].abi.as_ref().unwrap().functions[name][0];
-        if func.outputs.is_empty() {
+        if let Some(ret) = &instruction.returns {
+            let mut offset: usize = 0;
+            Some(decode_at_offset(
+                return_data,
+                &mut offset,
+                ret,
+                &self.stack[0].idl.as_ref().unwrap().types,
+            ))
```

Comment:
we should check there is no trailing data, iow `assert_eq!(return_data.len(), offset);`